### PR TITLE
MF::check_vector_compatibility(): improve assert message

### PR DESCRIPTION
**Pull request by peterrum**
FYI @nmuch @mschreter

### CODE REVIEW

File: include/deal.II/matrix_free/vector_access_internal.h [hunk 1]
```diff
@@ -201,8 +201,11 @@ namespace internal
            ExcMessage(
              "The parallel layout of the given vector is not "
              "compatible with the parallel partitioning in MatrixFree. "
-             "Use MatrixFree::initialize_dof_vector to get a "
-             "compatible vector."));
+             "A potential reason is that you did not use "
+             "MatrixFree::initialize_dof_vector() to get a "
+             "compatible vector. If you did, the dof_handler_index "
+             "used there and passed to the constructor of "
```
Comment (kronbichler) [inline include/deal.II/matrix_free/vector_access_internal.h:207]:
```suggestion
             "used there and the one passed to the constructor of "
```

**Review by kronbichler — APPROVED** [review]
If it could help, we might also be able to give the size of the vector and the number of ghosts as hints. But I think the error message is clearly improved already.

File: include/deal.II/matrix_free/vector_access_internal.h (hunk 1 above)
Comment (peterrum) [reply include/deal.II/matrix_free/vector_access_internal.h]: I think this should be enough!

**Review by kronbichler — APPROVED** [review]

**Pull request merged**